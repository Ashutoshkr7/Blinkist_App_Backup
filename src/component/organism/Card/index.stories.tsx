import Card from '.';
import {ComponentMeta, ComponentStory} from '@storybook/react';
import image from '../../../assets/book.png';

const CardComponent = {
    title : 'organism/Card',
    component: Card,
    parameters: {
        actions: {
          handles: ['mouseover'],
        },
    },
} as ComponentMeta<typeof Card>;

export const Template: ComponentStory<typeof Card>  = (args:any) => <Card {...args} />


export const Library = Template.bind({});

Library.args = {
    imgHeight: 300,
    url: image,
    bookName: "Being Boss",
    writerName: "Kathleen Shannon and Emily...",
    timeRead: "13-minute read",
    width: 350,
    progress: 60,
    inLibrary: true,
}


export const NotInLibrary = Template.bind({})

NotInLibrary.args = {
    imgHeight: 300,
    url: image,
    bookName: "Being Boss",
    writerName: "Kathleen Shannon and Emily...",
    timeRead: "13-minute read",
    width: 350,
    inLibrary: false,
}

export default CardComponent;