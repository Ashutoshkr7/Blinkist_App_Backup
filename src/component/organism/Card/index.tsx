import {Paper, Box} from '@mui/material';
import Image from '../../atom/Image';
import {AccessAlarm, MoreHoriz} from '@mui/icons-material'
import Typography from '../../atom/Typography';
import ProgressBar from '../../atom/ProgressBar';
import Icon from '../../atom/Icon';
import AddIcon from '@mui/icons-material/Add';
const CardComponent = ({imgHeight, url, bookName, writerName, timeRead = "0-minute read", progress, width, inLibrary }: any) => {
    return(
        <Paper elevation={1} sx={{borderRadius: '10px', width: `${width}px`, margin: "10px"}}>
            <Image sx={{borderRadius: '10px 10px 0px 0px'}} height={imgHeight} width={width} alt="blinkist" component="img"  src={url}/>
            <Box sx={{padding: '5px'}}>
                <Typography variant="h5" sx={{fontWeight: 'bold'}}>
                    {bookName}
                </Typography>
                <Typography variant="h6" sx={{fontWeight: '800', color: 'gray'}}>
                    {writerName}
                </Typography>
                <Box 
                    sx={{display: 'flex',
                    alignItems: 'center'
                    }}
                >
                    <Icon sx={{padding: '6px'}} icon={<AccessAlarm />} />
                    <Typography sx={{color: "gray", fontSize: "16px" }} variant="body">
                        {timeRead}
                    </Typography>
                </Box>
                <Box 
                    sx={{display: 'flex',
                        justifyContent: 'end'
                    }}
                >
                    <Icon sx={{padding: '6px'}} icon={<MoreHoriz />} />
                </Box>
            </Box>
            {inLibrary ? 
                <ProgressBar sx={{borderRadius: '0px 0px 10px 10px'}} width={100} value={progress} color="success"/> :
                <Box sx={{
                    display: 'flex', 
                    justifyContent: 'center', 
                    alignItems: 'center', 
                    color: "#0365F2",
                    height: '60px',
                    borderTop: '1px solid lightGrey'
                }}>
                    <Icon variant="h4" icon={<AddIcon sx={{fontWeight: 'bold', fontSize: '35px'}}/>} />
                    <Typography variant="h6" sx={{fontWeight: 'bold'}}>
                        Add to library 
                    </Typography>
                </Box>
            }
        </Paper>
    );
}

export default CardComponent; 