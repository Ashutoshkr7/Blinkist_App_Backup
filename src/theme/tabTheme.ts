import { createTheme } from '@mui/material/styles';
import { fontWeight } from '@mui/system';

const tabsTheme = createTheme({
    palette: {
      secondary: { 
        main: '#2CE080'
      }
    }
});

export default tabsTheme;