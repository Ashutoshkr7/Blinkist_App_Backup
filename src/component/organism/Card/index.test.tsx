import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '.';
import theme from '../../../theme/mainTheme'
import {BrowserRouter} from 'react-router-dom';
const props = {
    imgHeight: 300,
    url: '/assets/book.png',
    bookName: "Being Boss",
    writerName: "Kathleen Shannon and Emily...",
    timeRead: "13-minute read",
    width: 350,
    inLibrary: false,
}

const router = ()=>{
    return (
        <BrowserRouter>
            <Card {...props}></Card>
        </BrowserRouter>
    )
}

test('should render same bookname text', () => {
  render(router());
  const linkElement = screen.getByText('Being Boss');
  expect(linkElement).toBeInTheDocument();
});


test('should render same writer name text', () => {
    render(router());
    const linkElement = screen.getByText('Kathleen Shannon and Emily...');
    expect(linkElement).toBeInTheDocument();
});
  
test('should render same time read text', () => {
    render(router());
    const linkElement = screen.getByText('13-minute read');
    expect(linkElement).toBeInTheDocument();
});
  