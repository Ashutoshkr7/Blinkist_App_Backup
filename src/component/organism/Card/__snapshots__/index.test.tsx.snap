// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-v47ndi-MuiPaper-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "borderRadius": "8px",
      "margin": "16px 16px 16px",
    }
  }
>
  <a
    href="/book-info/undefined"
    onClick={[Function]}
  >
    <img
      alt="not found"
      className="MuiBox-root css-bohtpt"
      src="/assets/book.png"
      style={
        Object {
          "borderRadius": "8px 8px 0 0",
        }
      }
    />
  </a>
  <div
    className="MuiBox-root css-cxmp86"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-ly9sr8-MuiTypography-root"
      style={
        Object {
          "fontSize": "18px",
          "fontWeight": "bold",
        }
      }
    >
      Being Boss
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 css-12vk82c-MuiTypography-root"
    >
      Kathleen Shannon and Emily...
    </p>
    <div
      className="MuiBox-root css-0"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <div
        className="MuiBox-root css-old1by"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccessAlarmIcon"
          focusable="false"
          style={
            Object {
              "color": "#6D787E",
              "fontSize": "20px",
              "marginTop": "4px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="m22 5.72-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39 6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption2 css-1fo51um-MuiTypography-root"
      >
        13-minute read
      </span>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      
    </div>
  </div>
  <div
    className="MuiBox-root css-79elbk"
  >
    <div
      className="MuiBox-root css-1ah5qgj"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": "0 0 8px 8px",
          "borderTop": "1px solid lightGrey",
          "display": "flex",
          "height": "48px",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="MuiBox-root css-old1by"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-t5ymw4-MuiTypography-root"
      >
        Add to library
      </p>
    </div>
  </div>
</div>
`;
